catkin_ws/src/planner/src/missions.py

```diff
@@ -168,6 +168,10 @@ def pinger(
                 control=self.control,
                 state=self.state,
                 mapping=self.mapping,
+                goal_object="Buoy",
```

Comment:
the goal object should not be hardcoded. You can check the value of mission_after_success and then assign the goal object. For your comp, it should only be octagon. I also prefer not hardcoding the frequency.